internal/graph/storagewrapper.go

```diff
@@ -72,6 +72,10 @@ type CachedDatastore struct {
 	// sf is used to prevent draining the same iterator
 	// across multiple requests.
 	sf *singleflight.Group
+
+	// wg is used to synchronize inflight goroutines from underlying
+	// cached iterators.
+	wg *sync.WaitGroup
```

Comment:
Two questions:

1. How come this is a pointer?
2. What's our stance on short-handing things like this, similar to how we're short-handing `storage.OpenFGADatastore` on line 65?

If this was an actual piece of data without the `wg` key, it'd remove the need for all the additions in the test file

Reply:
> similar to how we're short-handing storage.OpenFGADatastore on line 65?

That line is an example of struct embedding, which allows `CachedDatastore` to automatically inherit all the methods in the OpenFGADatastore interface.

Reply:
Ah that is the official name, thank you. Is there a reason not to do it with WaitGroup here also? Is it just clarity? If this definition looked more like
```
type CachedDatastore struct {
  ...other stuff
  sync.WaitGroup
}
```
We could remove the need to pass in a `&WaitGroup{}` pointer at the time of construction. I don't know if that would be an anti-pattern, since I'm not a Go expert, am just curious why we went this way here.

Also, unrelated to the struct embedding question, how come we use a pointer type here rather than the actual object?

Reply:
> Is there a reason not to do it with WaitGroup here also? 

Yes, because you don't want to be able to expose waitgroup's methods, i.e. `ds.Group.Do`, `ds.Group.Forget` and `ds.Group.DoChan`.

Reply:
> how come we use a pointer type here rather than the actual object?
in this case specifically, `sync.WaitGroup` (and most primitives exposed in `sync`) are not safe to be copied around. See: https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/sync/waitgroup.go;l=13-30

Reply:
That all makes sense, thank you